edgedb/protocol/codecs/codecs.pyx

```diff
@@ -381,6 +379,36 @@ cdef duration_decode(pgproto.CodecContext settings, FRBuffer *buf):
     return datatypes.new_duration(microseconds, days, months)
 
 
+cdef decimal_decode(pgproto.CodecContext settings, FRBuffer *buf):
+    return pgproto.numeric_decode_binary_ex(settings, buf, True)
+
+
+cdef bigint_encode(pgproto.CodecContext settings, WriteBuffer buf, obj):
+    if isinstance(obj, float):
+        raise TypeError(
+            f'expected an integer or an integral Decimal'
+        )
+    elif isinstance(obj, decimal.Decimal):
+        if obj.as_tuple().exponent < 0:
+            raise TypeError(
+                f'expected an integer or an integral Decimal'
+            )
+    elif not isinstance(obj, int):
+        try:
+            obj = int(obj)
```

Comment:
This means we also accept strings. Compare this with the check we have for other ints:

```
        if type(obj) is not int and hasattr(type(obj), '__int__'):
```

You should do the same thing here.

Reply:
Fixed